packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,225 @@
+| `title`  | `string`        | `undefined` | Title of the BottomSheet footer      | ✅       |
+| `prefix` | `IconComponent` | `undefined` | Prefix icon placed before title text |          |
+
+### `BottomSheetFooterTrailing` API
+
+| Prop      | Type                                     | Default     | Description                                | Required |
+| --------- | ---------------------------------------- | ----------- | ------------------------------------------ | -------- |
+| `actions` | `{ primary: Action, secondary: Action }` | `undefined` | Renders a primary/secondary action buttons |          |
+
+```ts
+type Action = {
+  onClick: () => void;
+  text: string;
+  type?: 'button' | 'reset' | 'submit';
+  accessibilityLabel: string;
+  isLoading: boolean;
+  isDisabled: boolean;
+  icon: React.ReactNode;
+  iconPosition: 'left' | 'right';
+};
+```
+
+## Composition with DropDown
+
+We will export `BottomSheet` component separately as an independant component but generally the pattern will be to use it with SelectInput, where in mobile devices the Select's dropdown will be replaced by the BottomSheet.
+
+There are two approaches to doing it:
+
+1. We coupled the BottomSheet & SelectInput tightly and internally conditionally switch the components
+
+Pros:
+
+- Easy implementation for user's end
+- User's don't have to think about breakpoints, conditional rendering etc
+- From blade side, we will have greater control over the pattern
+
+Cons:
+
+- Bundle size will be impacted, even if users are desktop they will get the bundle of BottomSheet (vice versa)
+
+2. We expose BottomSheet indepandantly and let users lazy load the component as needed
+
+Pros:
+
+- No uneccesary bundle size impact for any of the platforms
+
+Cons:
+
+- Not trivial to implement from user's end, they will have to compose the BottomSheet & Select as per their needs.
+
+
+Considering the bundle size downside to approach 1, we decided to go ahead with approach 2.
+
+### Composition Example
+
+```jsx
+const BottomSheet = React.lazy();
+const DropdownOverlay = React.lazy();
+
+const SelectContent = () => {
+  return (
+    <>
+      <ActionListItem
+        leading={<ActionListItemIcon icon={SettingsIcon} />}
+        title="Settings"
+        value="settings"
+      />
+      <ActionListItem leading={<ActionListItemIcon icon={InfoIcon} />} title="Info" value="info" />
+    </>
+  );
+};
+
+const App = () => {
+  const [open, setOpen] = useState(false);
+
+  return (
+    <Dropdown selectionType={selectionType}>
+      <SelectInput label="Select Action" onClick={() => setOpen(true)} />
+      {isMobile ? (
+        <BottomSheet open={open} onDismiss={() => setOpen(false)}>
+          <BottomSheetHeader>
+            <BottomSheetHeaderLeading title="Payments Links" prefix={PayIcon} />
+          </BottomSheetHeader>
+          <BottomSheetBody>
+            <SelectContent />
+          </BottomSheetBody>
```

Comment:
Can we use ActionList component instead of BottomSheetBody?

```jsx
<ActionList>
  <SelectContent />
</ActionList>
```

Reply:
Yes, but functionality might [not](https://github.com/razorpay/blade/pull/1002#discussion_r1112791644) work. I need to check this. Since it's tied to the dropdown a bit. 

Also the <ActionList \/>'s root styling also not needed in the bottomsheet. (it has some padding around it so I might need to create a BaseActionList for functionality)